### ISSUE TITLE
Regression observed since 2023.2: NVDA freezes when switching windows

### ISSUE THREAD

**Issue by cary-rowen** [nvaccess/nvda#15920]
### Steps to reproduce:
There may not be exact steps to reproduce at this time.
Switching between windows that contain editable controls, such as Notepad++ or Notepad, can easily trigger freezing.
Restarting NVDA by pressing Ctrl+Alt+N is also very slow in these applications, taking 5 to 10 seconds.
### Actual behavior:
NVDA freezes and becomes unresponsive for a long time.

### Expected behavior:
NVDA should not freeze like it did in 2023.1.
### NVDA logs, crash dumps and other attachments:
[freeze.log](https://github.com/nvaccess/nvda/files/13663493/freeze.log)

#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
2023.3,2023.2 and Alpha
#### Windows version:
Windows version: Windows 11 23H2 (10.0.22631) workstation AMD64

#### Name and version of other software in use when reproducing the issue:
NVDA
#### Other information about your system:
None
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
2023.1 works as expected
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

**Comment by cary-rowen** [comment]
Several users in the Chinese community have reported this issue, and I'd love to hear more voices related to this.

**Comment by cary-rowen** [comment]
Please note that this issue still exists in 2024.1beta1.

**Comment by cary-rowen** [comment]
cc @seanbudd @jcsteh

**Comment by Adriani90** [comment]
@cary-rowen do the users have a braille display enabled?

And what kind of synthesizer are they using? eSpeak? Sapi4? Onecore?

Here some PRs that could have had an impact on this cc: @LeonarddeR, @mzanm 
#15295, #14924, #14759, #14588.

**Comment by cary-rowen** [comment]
This issue has nothing to do with the synthesizer, the other party is not using a braille display.We are restoring PRs one by one for testing.

**Comment by Adriani90** [comment]
@cary-rowen do you have any updates on this? Have you found any new results after testing with or without any of the referenced PRs above?

**Comment by hwf1324** [comment]
The problem seems to be caused by, among other things, the co-existence of an unquit `nvda.exe` and a new NVDA.

Here is the information I investigated remotely on the user's computer:

When NVDA is running:

When restarting NVDA or opening a new NVDA instance directly: the previous NVDA instance was terminated normally.

When exiting NVDA directly:
2022.4 or 2023.1: `nvda.exee` process exit
2023.2 and higher: `nvda.exe` process, exits after a long time or stays on. Starting a new NVDA does not cause the previous `nvda.exee` to exit.

[nvda-2023.1.log](https://github.com/nvaccess/nvda/files/13998381/nvda-2023.1.log):

```log
DEBUG - core._terminate (20:01:11.553) - MainThread (14136):
Terminating hwIo
DEBUG - core._terminate (20:01:11.553) - MainThread (14136):
Terminating addonHandler
DEBUG - core._terminate (20:01:11.553) - MainThread (14136):
Terminating garbageHandler
DEBUG - core.main (20:01:12.606) - MainThread (14136):
core done
INFO - __main__ (20:01:12.808) - MainThread (14136):
NVDA exit
```

[nvda-2023.1-Logfile.CSV](https://github.com/nvaccess/nvda/files/13998385/nvda-2023.1-Logfile.CSV):

```csv
"20:01:11.5493305","nvda.exe","14220","Thread Exit","","SUCCESS","Thread ID: 14868, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:01:11.5496737","nvda.exe","14220","Thread Exit","","SUCCESS","Thread ID: 13644, User Time: 0.0156250, Kernel Time: 0.0000000"
"20:01:11.5500585","nvda.exe","14220","Thread Exit","","SUCCESS","Thread ID: 10004, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:01:11.5532652","nvda.exe","14220","Thread Exit","","SUCCESS","Thread ID: 13796, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:01:11.6086584","nvda.exe","14220","Process Profiling","","SUCCESS","User Time: 0.7968750 seconds, Kernel Time: 0.6406250 seconds, Private Bytes: 42,188,800, Working Set: 74,190,848"
"20:01:12.6231064","nvda.exe","14220","Process Profiling","","SUCCESS","User Time: 0.8125000 seconds, Kernel Time: 0.6406250 seconds, Private Bytes: 42,188,800, Working Set: 74,199,040"
"20:01:12.8398485","nvda.exe","14220","Thread Exit","","SUCCESS","Thread ID: 13496, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:01:12.8599783","nvda.exe","14220","Thread Exit","","SUCCESS","Thread ID: 13776, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:01:12.8599819","nvda.exe","14220","Thread Exit","","SUCCESS","Thread ID: 14568, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:01:12.8599856","nvda.exe","14220","Thread Exit","","SUCCESS","Thread ID: 11480, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:01:12.8599877","nvda.exe","14220","Thread Exit","","SUCCESS","Thread ID: 14372, User Time: 0.0625000, Kernel Time: 0.0156250"
"20:01:12.8599907","nvda.exe","14220","Thread Exit","","SUCCESS","Thread ID: 14160, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:01:12.8599914","nvda.exe","14220","Thread Exit","","SUCCESS","Thread ID: 13572, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:01:12.8600010","nvda.exe","14220","Thread Exit","","SUCCESS","Thread ID: 15204, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:01:12.8600013","nvda.exe","14220","Thread Exit","","SUCCESS","Thread ID: 11620, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:01:12.8600060","nvda.exe","14220","Thread Exit","","SUCCESS","Thread ID: 14652, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:01:12.8600068","nvda.exe","14220","Thread Exit","","SUCCESS","Thread ID: 14932, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:01:12.8600092","nvda.exe","14220","Thread Exit","","SUCCESS","Thread ID: 14908, User Time: 0.0156250, Kernel Time: 0.0000000"
"20:01:12.8600275","nvda.exe","14220","Thread Exit","","SUCCESS","Thread ID: 276, User Time: 0.0000000, Kernel Time: 0.0156250"
"20:01:12.8643652","nvda.exe","14220","Thread Exit","","SUCCESS","Thread ID: 14136, User Time: 0.6562500, Kernel Time: 0.5312500"
"20:01:12.8661114","nvda.exe","14220","Process Exit","","SUCCESS","Exit Status: 0, User Time: 0.8437500 seconds, Kernel Time: 0.6562500 seconds, Private Bytes: 38,277,120, Peak Private Bytes: 46,968,832, Working Set: 70,836,224, Peak Working Set: 93,655,040"
```

[nvda-2023.2.log](https://github.com/nvaccess/nvda/files/13998383/nvda-2023.2.log):

```log
DEBUG - core._terminate (20:06:36.085) - MainThread (14184):
Terminating bdDetect
DEBUG - core._terminate (20:06:36.085) - MainThread (14184):
Terminating hwIo
DEBUG - core._terminate (20:06:36.086) - MainThread (14184):
Terminating addonHandler
DEBUG - core._terminate (20:06:36.086) - MainThread (14184):
Terminating garbageHandler
DEBUG - core._terminate (20:06:37.136) - MainThread (14184):
Terminating NVDAHelper
DEBUG - core.main (20:06:37.140) - MainThread (14184):
core done
INFO - __main__ (20:06:37.341) - MainThread (14184):
NVDA exit
```

[nvda-2023.2-Logfile.CSV](https://github.com/nvaccess/nvda/files/13998401/nvda-2023.2-Logfile.CSV):

```csv
"20:06:35.9741929","nvda.exe","12084","Thread Exit","","SUCCESS","Thread ID: 1620, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:06:35.9742214","nvda.exe","12084","Thread Exit","","SUCCESS","Thread ID: 11020, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:06:35.9742967","nvda.exe","12084","Thread Exit","","SUCCESS","Thread ID: 14500, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:06:35.9904862","nvda.exe","12084","Thread Exit","","SUCCESS","Thread ID: 3124, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:06:36.0168011","nvda.exe","12084","Thread Create","","SUCCESS","Thread ID: 13484"
"20:06:36.0170035","nvda.exe","12084","Thread Create","","SUCCESS","Thread ID: 5152"
"20:06:36.0781099","nvda.exe","12084","Thread Exit","","SUCCESS","Thread ID: 15168, User Time: 0.0156250, Kernel Time: 0.0937500"
"20:06:36.0813636","nvda.exe","12084","Thread Exit","","SUCCESS","Thread ID: 3812, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:06:36.0820093","nvda.exe","12084","Thread Exit","","SUCCESS","Thread ID: 13576, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:06:36.0821356","nvda.exe","12084","Thread Exit","","SUCCESS","Thread ID: 14264, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:06:36.0857691","nvda.exe","12084","Thread Exit","","SUCCESS","Thread ID: 2424, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:06:36.5591205","nvda.exe","12084","Process Profiling","","SUCCESS","User Time: 0.7656250 seconds, Kernel Time: 0.9843750 seconds, Private Bytes: 45,793,280, Working Set: 79,327,232"
"20:06:36.5591208","nvdaHelperRemoteLoader.exe","3628","Process Profiling","","SUCCESS","User Time: 0.0000000 seconds, Kernel Time: 0.0156250 seconds, Private Bytes: 1,466,368, Working Set: 6,627,328"
"20:06:37.1368195","nvda.exe","12084","Thread Exit","","SUCCESS","Thread ID: 6096, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:06:37.1372399","nvdaHelperRemoteLoader.exe","3628","Load Image","C:\Users\Administrator\AppData\Local\Temp\64F3.tmp","SUCCESS","Image Base: 0x7ffb83870000, Image Size: 0x22000"
"20:06:37.1381144","nvdaHelperRemoteLoader.exe","3628","Load Image","C:\Windows\System32\cryptbase.dll","SUCCESS","Image Base: 0x7ffbaab10000, Image Size: 0xc000"
"20:06:37.1385083","nvdaHelperRemoteLoader.exe","3628","Thread Exit","","SUCCESS","Thread ID: 14976, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:06:37.1391385","nvdaHelperRemoteLoader.exe","3628","Load Image","C:\Windows\System32\kernel.appcore.dll","SUCCESS","Image Base: 0x7ffba9a50000, Image Size: 0x12000"
"20:06:37.1392707","nvdaHelperRemoteLoader.exe","3628","Thread Exit","","SUCCESS","Thread ID: 14532, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:06:37.1392747","nvdaHelperRemoteLoader.exe","3628","Thread Exit","","SUCCESS","Thread ID: 492, User Time: 0.0000000, Kernel Time: 0.0156250"
"20:06:37.1392904","nvdaHelperRemoteLoader.exe","3628","Thread Exit","","SUCCESS","Thread ID: 9816, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:06:37.1395943","nvdaHelperRemoteLoader.exe","3628","Thread Exit","","SUCCESS","Thread ID: 11000, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:06:37.1399897","nvdaHelperRemoteLoader.exe","3628","Process Exit","","SUCCESS","Exit Status: 0, User Time: 0.0000000 seconds, Kernel Time: 0.0156250 seconds, Private Bytes: 1,478,656, Peak Private Bytes: 1,531,904, Working Set: 7,237,632, Peak Working Set: 7,249,920"
"20:06:37.1582850","nvda.exe","12084","Thread Exit","","SUCCESS","Thread ID: 15320, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:06:37.5580139","nvda.exe","12084","Process Profiling","","SUCCESS","User Time: 0.7656250 seconds, Kernel Time: 1.0156250 seconds, Private Bytes: 45,568,000, Working Set: 79,294,464"
"20:06:38.5572473","nvda.exe","12084","Process Profiling","","SUCCESS","User Time: 0.7656250 seconds, Kernel Time: 1.0156250 seconds, Private Bytes: 45,834,240, Working Set: 79,368,192"
"20:06:39.5601280","nvda.exe","12084","Process Profiling","","SUCCESS","User Time: 0.7656250 seconds, Kernel Time: 1.0156250 seconds, Private Bytes: 45,834,240, Working Set: 79,446,016"
"20:06:40.5597446","nvda.exe","12084","Process Profiling","","SUCCESS","User Time: 0.7656250 seconds, Kernel Time: 1.0156250 seconds, Private Bytes: 45,834,240, Working Set: 79,446,016"
...
"20:18:08.5577434","nvda.exe","12084","Process Profiling","","SUCCESS","User Time: 0.7812500 seconds, Kernel Time: 1.0156250 seconds, Private Bytes: 45,584,384, Working Set: 79,736,832"
"20:18:09.5655215","nvda.exe","12084","Process Profiling","","SUCCESS","User Time: 0.7812500 seconds, Kernel Time: 1.0156250 seconds, Private Bytes: 45,584,384, Working Set: 79,736,832"
"20:18:10.5514804","nvda.exe","12084","Process Profiling","","SUCCESS","User Time: 0.7812500 seconds, Kernel Time: 1.0156250 seconds, Private Bytes: 45,584,384, Working Set: 79,736,832"
"20:18:10.9003679","nvda.exe","12084","Thread Exit","","SUCCESS","Thread ID: 14344, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:18:10.9003739","nvda.exe","12084","Thread Exit","","SUCCESS","Thread ID: 1900, User Time: 0.0468750, Kernel Time: 0.0312500"
"20:18:10.9003764","nvda.exe","12084","Thread Exit","","SUCCESS","Thread ID: 9576, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:18:10.9003833","nvda.exe","12084","Thread Exit","","SUCCESS","Thread ID: 15028, User Time: 0.0156250, Kernel Time: 0.0156250"
"20:18:10.9003871","nvda.exe","12084","Thread Exit","","SUCCESS","Thread ID: 15164, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:18:10.9003875","nvda.exe","12084","Thread Exit","","SUCCESS","Thread ID: 14384, User Time: 0.0000000, Kernel Time: 0.0468750"
"20:18:10.9003884","nvda.exe","12084","Thread Exit","","SUCCESS","Thread ID: 15080, User Time: 0.0156250, Kernel Time: 0.0156250"
"20:18:10.9003893","nvda.exe","12084","Thread Exit","","SUCCESS","Thread ID: 14184, User Time: 0.6562500, Kernel Time: 0.6562500"
"20:18:10.9003903","nvda.exe","12084","Thread Exit","","SUCCESS","Thread ID: 13904, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:18:10.9003988","nvda.exe","12084","Thread Exit","","SUCCESS","Thread ID: 14916, User Time: 0.0000000, Kernel Time: 0.0156250"
"20:18:10.9026787","nvda.exe","12084","Process Exit","","SUCCESS","Exit Status: 1, User Time: 0.7812500 seconds, Kernel Time: 1.0312500 seconds, Private Bytes: 45,547,520, Peak Private Bytes: 50,364,416, Working Set: 79,077,376, Peak Working Set: 98,725,888"
```

Here's where I forcibly killed `nvda.exe`.

[nvda-2023.3.1.log](https://github.com/nvaccess/nvda/files/13998417/nvda-2023.3.1.log)

Here for some reason `nvda-old.log` disappeared

By the way, the exit window is not displayed in the foreground at `NVDA+q`.

[nvda-2023.3.1-Logfile.CSV](https://github.com/nvaccess/nvda/files/13998416/nvda-2023.3.1-Logfile.CSV):

```csv
"20:23:33.0349520","nvda.exe","12548","Thread Exit","","SUCCESS","Thread ID: 3624, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:23:33.0358513","nvdaHelperRemoteLoader.exe","14796","Load Image","C:\Users\Administrator\AppData\Local\Temp\64F3.tmp","SUCCESS","Image Base: 0x7ffb83870000, Image Size: 0x22000"
"20:23:33.0367875","nvdaHelperRemoteLoader.exe","14796","Load Image","C:\Windows\System32\cryptbase.dll","SUCCESS","Image Base: 0x7ffbaab10000, Image Size: 0xc000"
"20:23:33.0371826","nvdaHelperRemoteLoader.exe","14796","Thread Exit","","SUCCESS","Thread ID: 10100, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:23:33.0374671","nvdaHelperRemoteLoader.exe","14796","Load Image","C:\Windows\System32\kernel.appcore.dll","SUCCESS","Image Base: 0x7ffba9a50000, Image Size: 0x12000"
"20:23:33.0376109","nvdaHelperRemoteLoader.exe","14796","Thread Exit","","SUCCESS","Thread ID: 14916, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:23:33.0376116","nvdaHelperRemoteLoader.exe","14796","Thread Exit","","SUCCESS","Thread ID: 6112, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:23:33.0376126","nvdaHelperRemoteLoader.exe","14796","Thread Exit","","SUCCESS","Thread ID: 15088, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:23:33.0379422","nvdaHelperRemoteLoader.exe","14796","Thread Exit","","SUCCESS","Thread ID: 10004, User Time: 0.0156250, Kernel Time: 0.0000000"
"20:23:33.0383468","nvdaHelperRemoteLoader.exe","14796","Process Exit","","SUCCESS","Exit Status: 0, User Time: 0.0156250 seconds, Kernel Time: 0.0000000 seconds, Private Bytes: 1,458,176, Peak Private Bytes: 1,511,424, Working Set: 7,221,248, Peak Working Set: 7,233,536"
"20:23:33.9842284","nvda.exe","12548","Process Profiling","","SUCCESS","User Time: 1.4375000 seconds, Kernel Time: 0.7968750 seconds, Private Bytes: 50,495,488, Working Set: 82,505,728"
"20:23:34.9757415","nvda.exe","12548","Process Profiling","","SUCCESS","User Time: 1.4375000 seconds, Kernel Time: 0.7968750 seconds, Private Bytes: 50,495,488, Working Set: 82,505,728"
"20:23:35.9725801","nvda.exe","12548","Process Profiling","","SUCCESS","User Time: 1.4375000 seconds, Kernel Time: 0.7968750 seconds, Private Bytes: 50,495,488, Working Set: 82,505,728"
"20:23:36.9844799","nvda.exe","12548","Process Profiling","","SUCCESS","User Time: 1.4375000 seconds, Kernel Time: 0.7968750 seconds, Private Bytes: 50,495,488, Working Set: 82,505,728"
"20:23:37.2126382","nvda.exe","14576","Process Start","","SUCCESS","Parent PID: 7280, Command line: ""C:\NVDA\2023.3.1\nvda.exe"" , Current directory: C:\NVDA\2023.3.1\, Environment: 
;	=::=::\
;	ALLUSERSPROFILE=C:\ProgramData
;	APPDATA=C:\Users\Administrator\AppData\Roaming
;	CommonProgramFiles=C:\Program Files\Common Files
;	CommonProgramFiles(x86)=C:\Program Files (x86)\Common Files
;	CommonProgramW6432=C:\Program Files\Common Files
;	COMPUTERNAME=XZ-LG3F6HC0LGOT
;	ComSpec=C:\Windows\system32\cmd.exe
;	DriverData=C:\Windows\System32\Drivers\DriverData
;	HOMEDRIVE=C:
;	HOMEPATH=\Users\Administrator
;	LOCALAPPDATA=C:\Users\Administrator\AppData\Local
;	LOGONSERVER=\\XZ-LG3F6HC0LGOT
;	NUMBER_OF_PROCESSORS=16
;	OneDrive=D:\OneDrive
;	OneDriveConsumer=D:\OneDrive
;	OS=Windows_NT
;	Path=C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Scripts\;C:\Users\Administrator\AppData\Local\Programs\Python\Python311\;C:\Users\Administrator\AppData\Local\Microsoft\WindowsApps;
;	PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC
;	PROCESSOR_ARCHITECTURE=AMD64
;	PROCESSOR_IDENTIFIER=AMD64 Family 25 Model 116 Stepping 1, AuthenticAMD
;	PROCESSOR_LEVEL=25
;	PROCESSOR_REVISION=7401
;	ProgramData=C:\ProgramData
;	ProgramFiles=C:\Program Files
;	ProgramFiles(x86)=C:\Program Files (x86)
;	ProgramW6432=C:\Program Files
;	PSModulePath=C:\Program Files\WindowsPowerShell\Modules;C:\Windows\system32\WindowsPowerShell\v1.0\Modules
;	PUBLIC=C:\Users\Public
;	SESSIONNAME=Console
;	SystemDrive=C:
;	SystemRoot=C:\Windows
;	TEMP=C:\Users\ADMINI~1\AppData\Local\Temp
;	TMP=C:\Users\ADMINI~1\AppData\Local\Temp
;	USERDOMAIN=XZ-LG3F6HC0LGOT
;	USERDOMAIN_ROAMINGPROFILE=XZ-LG3F6HC0LGOT
;	USERNAME=Administrator
;	USERPROFILE=C:\Users\Administrator
;	windir=C:\Windows"
...
"20:23:47.9052887","nvda.exe","14576","Thread Exit","","SUCCESS","Thread ID: 14976, User Time: 0.0625000, Kernel Time: 0.0468750"
"20:23:47.9111860","nvda.exe","14576","Thread Exit","","SUCCESS","Thread ID: 12076, User Time: 0.0312500, Kernel Time: 0.0156250"
"20:23:47.9114301","nvda.exe","14576","Thread Exit","","SUCCESS","Thread ID: 14664, User Time: 0.0000000, Kernel Time: 0.0156250"
"20:23:47.9153027","nvda.exe","14576","Thread Exit","","SUCCESS","Thread ID: 8468, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:23:47.9782762","nvda.exe","14576","Process Profiling","","SUCCESS","User Time: 1.8125000 seconds, Kernel Time: 0.7656250 seconds, Private Bytes: 49,643,520, Working Set: 82,595,840"
"20:23:47.9782767","nvdaHelperRemoteLoader.exe","14032","Process Profiling","","SUCCESS","User Time: 0.0156250 seconds, Kernel Time: 0.0000000 seconds, Private Bytes: 1,462,272, Working Set: 6,623,232"
"20:23:48.9750387","nvda.exe","14576","Process Profiling","","SUCCESS","User Time: 1.8125000 seconds, Kernel Time: 0.7656250 seconds, Private Bytes: 49,643,520, Working Set: 82,595,840"
"20:23:48.9750390","nvdaHelperRemoteLoader.exe","14032","Process Profiling","","SUCCESS","User Time: 0.0156250 seconds, Kernel Time: 0.0000000 seconds, Private Bytes: 1,462,272, Working Set: 6,623,232"
"20:23:48.9874346","nvda.exe","14576","Thread Exit","","SUCCESS","Thread ID: 13664, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:23:48.9878547","nvdaHelperRemoteLoader.exe","14032","Load Image","C:\Users\Administrator\AppData\Local\Temp\64F3.tmp","SUCCESS","Image Base: 0x7ffb83870000, Image Size: 0x22000"
"20:23:48.9886885","nvdaHelperRemoteLoader.exe","14032","Load Image","C:\Windows\System32\cryptbase.dll","SUCCESS","Image Base: 0x7ffbaab10000, Image Size: 0xc000"
"20:23:48.9890695","nvdaHelperRemoteLoader.exe","14032","Thread Exit","","SUCCESS","Thread ID: 10020, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:23:48.9893445","nvdaHelperRemoteLoader.exe","14032","Load Image","C:\Windows\System32\kernel.appcore.dll","SUCCESS","Image Base: 0x7ffba9a50000, Image Size: 0x12000"
"20:23:48.9894665","nvdaHelperRemoteLoader.exe","14032","Thread Exit","","SUCCESS","Thread ID: 13996, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:23:48.9894722","nvdaHelperRemoteLoader.exe","14032","Thread Exit","","SUCCESS","Thread ID: 4100, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:23:48.9894896","nvdaHelperRemoteLoader.exe","14032","Thread Exit","","SUCCESS","Thread ID: 13776, User Time: 0.0000000, Kernel Time: 0.0000000"
"20:23:48.9897875","nvdaHelperRemoteLoader.exe","14032","Thread Exit","","SUCCESS","Thread ID: 11004, User Time: 0.0156250, Kernel Time: 0.0000000"
"20:23:48.9901764","nvdaHelperRemoteLoader.exe","14032","Process Exit","","SUCCESS","Exit Status: 0, User Time: 0.0156250 seconds, Kernel Time: 0.0000000 seconds, Private Bytes: 1,458,176, Peak Private Bytes: 1,511,424, Working Set: 7,221,248, Peak Working Set: 7,233,536"
"20:23:49.0287132","nvda.exe","14576","Thread Exit","","SUCCESS","Thread ID: 13620, User Time: 0.0000000, Kernel Time: 0.0156250"
"20:23:49.9823793","nvda.exe","14576","Process Profiling","","SUCCESS","User Time: 1.8125000 seconds, Kernel Time: 0.7812500 seconds, Private Bytes: 49,401,856, Working Set: 82,403,328"
"20:23:50.9869767","nvda.exe","14576","Process Profiling","","SUCCESS","User Time: 1.8125000 seconds, Kernel Time: 0.7812500 seconds, Private Bytes: 49,401,856, Working Set: 82,403,328"
"20:23:51.9828880","nvda.exe","14576","Process Profiling","","SUCCESS","User Time: 1.8125000 seconds, Kernel Time: 0.7812500 seconds, Private Bytes: 49,401,856, Working Set: 82,403,328"
"20:23:52.3592278","nvda.exe","14112","Process Start","","SUCCESS","Parent PID: 7280, Command line: ""C:\NVDA\2023.3.1\nvda.exe"" , Current directory: C:\NVDA\2023.3.1\, Environment: 
...
"20:24:07.2446168","nvda.exe","14576","Thread Create","","SUCCESS","Thread ID: 10004"
"20:24:07.9857837","nvda.exe","14576","Process Profiling","","SUCCESS","User Time: 1.8125000 seconds, Kernel Time: 0.7812500 seconds, Private Bytes: 49,545,216, Working Set: 82,452,480"
"20:24:07.9857848","nvda.exe","14112","Process Profiling","","SUCCESS","User Time: 1.4843750 seconds, Kernel Time: 0.9062500 seconds, Private Bytes: 48,373,760, Working Set: 81,735,680"
"20:24:08.9728857","nvda.exe","14576","Process Profiling","","SUCCESS","User Time: 1.8125000 seconds, Kernel Time: 0.7812500 seconds, Private Bytes: 49,545,216, Working Set: 82,452,480"
"20:24:08.9728868","nvda.exe","14112","Process Profiling","","SUCCESS","User Time: 1.4843750 seconds, Kernel Time: 0.9062500 seconds, Private Bytes: 48,373,760, Working Set: 81,735,680"
"20:24:09.9754419","nvda.exe","14576","Process Profiling","","SUCCESS","User Time: 1.8125000 seconds, Kernel Time: 0.7812500 seconds, Private Bytes: 49,545,216, Working Set: 82,452,480"
"20:24:09.9754430","nvda.exe","14112","Process Profiling","","SUCCESS","User Time: 1.4843750 seconds, Kernel Time: 0.9062500 seconds, Private Bytes: 48,373,760, Working Set: 81,735,680"
"20:24:10.9801474","nvda.exe","14576","Process Profiling","","SUCCESS","User Time: 1.8125000 seconds, Kernel Time: 0.7812500 seconds, Private Bytes: 49,545,216, Working Set: 82,452,480"
"20:24:10.9801485","nvda.exe","14112","Process Profiling","","SUCCESS","User Time: 1.4843750 seconds, Kernel Time: 0.9062500 seconds, Private Bytes: 48,373,760, Working Set: 81,735,680"
"20:24:11.9827860","nvda.exe","14576","Process Profiling","","SUCCESS","User Time: 1.8125000 seconds, Kernel Time: 0.7812500 seconds, Private Bytes: 49,545,216, Working Set: 82,452,480"
...
```

Please note the time and PID

In summary, there are multiple `nvda.` processes in the NVDA runtime.

The `logfile.csv` is a snippet of thread and Profiling events that I logged using the [Process Monitor](https://learn.microsoft.com/en-us/sysinternals/downloads/procmon) tool in the [Sysinternals suite](https://learn.microsoft.com/en-us/sysinternals/).
Using `... ` to omit some entries.

In the end, everything worked fine on my machine.

**Comment by hwf1324** [comment]
It appears that some network connectivity exists when NVDA exits. causing this issue.

**Comment by jcsteh** [comment]
FWIW, I've been seeing this issue for several months now (maybe up to a year), but I can't recall exactly when it started. 2023.2 would probably make sense, especially because I run alpha builds, so it would have been before 2023.2 was released. I always assumed it was some weird issue on my system given that I'm a developer and I tend to run bleeding edge builds of things, do a lot of process debugging, etc. It's pretty difficult to debug because the system is pretty much unusable when NVDA ends up in this state: many things stop responding, you can't switch applications properly, etc.

**Comment by jcsteh** [comment]
It's not great that the old process doesn't exit gracefully, but it's worse that we don't kill off the old process when we start the new copy. We do have code to do that, but it isn't working. My guess is that the message window has disappeared by the time we start the new copy. This makes me wonder whether the message window should be destroyed much later, though that might not be sufficient if we're getting stuck on a C++ thread.

**Comment by hwf1324** [comment]
This issue should be fixed by #16183

**Comment by cary-rowen** [comment]
Glad to see this issue resolved, I'm closing it now.
